e2e/helpers/browser_helper.js

```diff
@@ -100,6 +100,31 @@ module.exports = class BrowserHelpers extends Helper {
     }
   }
 
+  async waitForSpinnerToFinish() {
+    const helper = this.getHelper();
+    const waitTimeout = helper.options.waitForTimeout;
+    const spinnerElement = {css: 'xuilib-loading-spinner'};
+    const noCasesText = {xpath: '//div[contains(text(), "No cases found")]'};
+
+    let spinner = await this.hasSelector(spinnerElement);
+    if (await this.hasSelector(noCasesText) && !spinner) {
+      // short wait to allow spinner to appear
+      await helper.wait(3);
+    }
+    for (let i = 0; i <= waitTimeout; i++) {
+      spinner = await this.hasSelector(spinnerElement);
+      if (!spinner) {
+        break;
+      }
+      if (i < waitTimeout) {
+        console.log('ExUI loading spinner found, waiting...');
```

Comment:
The preferred way of logging in codecept is using their output object -> https://codecept.io/internal-api/#output

Reply:
You can see an example of usage in the steps_file.js

Reply:
good shout, I'll change this 👍